File: csharp/WorkOOP/Trapeze.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WorkOOP
{
    // Реализация класса Trapeze – трапеция.
    internal class Trapeze : QGon
    {
        /*
         * Trapeze – конструктор, описывающий инициализацию объекта. 
         * 
         * На вход принимает массив точек в порядке обхода по трапеции.
         */
        public Trapeze(Point2D[] p) : base(p) { }

        /*
         * Переопредение функции square под трапецию.
         */
        new public double square() => base.square();   
    }
}

File: python/src/core/trapeze.py
from core.qgon import QGon
from core.point2d import Point2D

class Trapeze(QGon):
	def __init__(self, p):
		self.n = 4
		self.p = p

	def __str__(self):
		st = f"Trapeze: "
		for i in range(len(self.p)):
			st += f"{str(self.p[i])}, "
		st = st[:len(st)-2]
		return st

	def square(self):
		return QGon(self.p).square()